tools/predict_net.py

```diff
@@ -0,0 +1,207 @@
+            self.cap.set(cv2.CAP_PROP_POS_FRAMES, 0)
+            frame = None
+
+        return was_read, frame
+
+    def clean(self):
+        self.cap.release()
+        cv2.destroyAllWindows()
+
+
+def predict(cfg):
+    # Set random seed from configs.
+    np.random.seed(cfg.RNG_SEED)
+    torch.manual_seed(cfg.RNG_SEED)
+
+    # Setup logging format.
+    logging.setup_logging()
+
+    # Print config.
+    logger.info("Predict with config:")
+    logger.info(cfg)
+    # Build the video model and print model statistics.
+    model = model_builder.build_model(cfg)
+    model.eval()
+    misc.log_model_info(model)
+
+   # Load a checkpoint to test if applicable.
+    if cfg.TEST.CHECKPOINT_FILE_PATH != "":
+        ckpt = cfg.TEST.CHECKPOINT_FILE_PATH
+        convert_from_caffe2 = cfg.TEST.CHECKPOINT_TYPE == "caffe2"
+    elif cu.has_checkpoint(cfg.OUTPUT_DIR):
+        ckpt = cfg.OUTPUT_DIR
+        convert_from_caffe2 = False
+    elif cfg.TRAIN.CHECKPOINT_FILE_PATH != "":
+        # If no checkpoint found in TEST.CHECKPOINT_FILE_PATH or in the current
+        # checkpoint folder, try to load checkpint from
+        # TRAIN.CHECKPOINT_FILE_PATH and test it.
+        ckpt = cfg.TRAIN.CHECKPOINT_FILE_PATH
+        convert_from_caffe2 = cfg.TRAIN.CHECKPOINT_TYPE == "caffe2"
+    else:
+        raise NotImplementedError("Unknown way to load checkpoint.")
+
+    cu.load_checkpoint(
+        ckpt,
+        model,
+        cfg.NUM_GPUS > 1,
+        None,
+        inflation=False,
+        convert_from_caffe2=convert_from_caffe2,
+    )
+
+    # TODO: Take inputs from camera proceed below, definitely use for loop.
+    # Load the labels of Kinectics-400 dataset
+    labels_df = pd.read_csv(cfg.PREDICT.LABEL_FILE_PATH)
+    labels = labels_df['name'].values
+    img_provider = VideoReader(cfg.PREDICT.SOURCE)
+    frames = []
+    label = 'predicting...'
+    s = 0.
+    i = -1
+    for able_to_read, frame in img_provider:
+        i += 1
+        if not able_to_read:
+            # when reaches the end frame, clear the buffer and continue to the next one.
+            frames = []
+            continue
+
+        if len(frames) != cfg.DATA.NUM_FRAMES*cfg.DATA.SAMPLING_RATE:
+            frame_processed = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+            frame_processed = imresize(frame_processed, 256)
+            frames.append(frame_processed)
+            
+        if len(frames) == cfg.DATA.NUM_FRAMES*cfg.DATA.SAMPLING_RATE:
+            start = time()
+            # Perform color normalization.
+            inputs = torch.tensor(frames).float()
+            inputs = inputs / 255.0
+            inputs = inputs - torch.tensor(cfg.DATA.MEAN)
+            inputs = inputs / torch.tensor(cfg.DATA.STD)
+            # T H W C -> C T H W.
+            inputs = inputs.permute(3, 0, 1, 2)
+            # 1 C T H W.
+            inputs = inputs[None, :, :, :, :]
+            # Sample frames for the fast pathway.
```

Comment:
You can also use the `pack_pathway_output` in `slowfast/datasets/loader.py`.
I could also refactor the pack_pathway_output later. :P